cpp/include/cuvs/preprocessing/spectral/spectral_embedding_types.hpp

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cstdint>
+
+namespace cuvs::preprocessing::spectral {
+
+struct spectral_embedding_config {
```

Comment:
cuVS is not header-only, so we no longer need an additional "types" header. Put the API definition and the params struct into the same header file please. We do this for RAFT because the params are lightweight but the header itself pulls in all the cuh headers- that way we have a lightweight header for being able to specify API vocabilary that's independent of the code that needs to be compiled w/ CUDA. Since cuVS is a formal C++ API (with source files) like cuML, we don't need this restriction. 

Please also rename this to `cuvs::preprocessing::spectral::params` so that it is consistent with the other APIs (no need to duplicate the "spectral_embedding_" since that's in the namespace).